src/minimal.rs

```diff
@@ -242,6 +257,12 @@ impl MinimalSecureLayer {
                 if hash != sha256(datas_hashed).as_ref() {
                     return Err(IncomingMsgErr::InvalidHashOrSig.into());
                 }
+
+                // Update nonce_list
+                self.nonce_list.insert(nonce);
+                while self.nonce_list.contains(&self.recv_nonce) {
```

Comment:
The list of orphan nonces will thus grow indefinitely, it's very bad for memory usage that, it is necessary to replace the contains() by a remove() followed by an is_some()